### PR TITLE
BACKLOG-12143 : use Dropdown moonstone component for siteswitcher and…

### PR DESCRIPTION
**Pull request by Yousria**
… language switcher

## JIRA

https://jira.jahia.org/browse/BACKLOG-12143

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/LanguageSwitcher.jsx [hunk 1]
```diff
@@ -5,15 +5,25 @@ import {useTranslation} from 'react-i18next';
...
 import {ProgressOverlay, withNotifications} from '@jahia/react-material';
 import {useSiteInfo} from '@jahia/data-helper';
-import {cmSetAvailableLanguages} from '../JContent.redux';
-import {LanguageSwitcher} from '@jahia/design-system-kit';
 import {registry} from '@jahia/ui-extender';
+import {Dropdown} from '@jahia/moonstone';
 
-export const SiteLanguageSwitcher = ({
+function createDataObjects(languages) {
+    let data = [];
```
Comment (tdraier) [inline src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/LanguageSwitcher.jsx]: Can't you use a map() for doing it in a single-line ? (and no need for a function, actually)

File: src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/LanguageSwitcher.jsx [hunk 2]
```diff
@@ -37,15 +47,15 @@ export const SiteLanguageSwitcher = ({
...
     return (
-        <LanguageSwitcher
-            lang={lang}
-            languages={siteInfo.languages}
-            color="inverted"
-            onSelectLanguage={lang => onSelectLanguageHandler(lang)}
+        <Dropdown
+            label={lang}
+            value={lang}
+            data={createDataObjects(siteInfo.languages)}
```
Comment (tdraier) [inline src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/LanguageSwitcher.jsx]: here it is
```suggestion
            data={siteInfo.languages.map(l => ({label: l.language, value: l.language}))}
```

File: src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/SiteSwitcher.jsx [hunk 3]
```diff
@@ -101,18 +115,20 @@ class SiteSwitcher extends React.Component {
...
                     let sites = this.getSites(data);
                     return (
-                        <SiteSwitcherDisplay
-                            siteKey={siteKey}
-                            siteNodes={sites}
-                            onSelectSite={siteNode => {
+                        <Dropdown
+                            label={siteKey}
+                            value={siteKey}
+                            data={createDataObjects(sites)}
```
Comment (tdraier) [inline src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/SiteSwitcher.jsx]: same as for language, use a map

Comment (Yousria) [reply src/javascript/JContent/AppLayout/ContentNavigation/NavigationHeader/SiteSwitcher.jsx]: done :)

**Pull request merged**